src/SonarQube.Client/Models/Connections.cs

```diff
@@ -20,17 +20,27 @@
 
 using System;
 using System.Security;
+using SonarQube.Client.Helpers;
 
-namespace SonarLint.VisualStudio.Integration.Service
+namespace SonarQube.Client.Models
 {
+    // TODO: I want to use this class but there are a lot changes to do so keep using the other now on.
+    //public class Connection
+    //{
+    //    public Uri ServerUri { get; set; }
+    //    public string Login { get; set; }
+    //    public SecureString Password { get; set; }
+    //    public AuthenticationType Authentication { get; set; }
+    //}
+
     /// <summary>
     /// Represents the connection information needed to connect to SonarQube service
     /// </summary>
-    internal class ConnectionInformation : ICloneable, IDisposable
+    public class ConnectionInformation : ICloneable, IDisposable
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Fix this implementation of 'IDisposable' to conform to the dispose pattern. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3881)